core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -628,12 +630,15 @@ public PlanNode plan(Update node)
         // The integer merge case number, always 0 for update
         Metadata metadata = plannerContext.getMetadata();
         ImmutableList.Builder<Expression> rowBuilder = ImmutableList.builder();
+        Assignments.Builder assignments = Assignments.builder();
 
         // Add column values to the rowBuilder - - the SET expression value for updated
         // columns, and the existing column value for non-updated columns
         for (int columnIndex = 0; columnIndex < mergeAnalysis.getDataColumnHandles().size(); columnIndex++) {
             ColumnHandle dataColumnHandle = mergeAnalysis.getDataColumnHandles().get(columnIndex);
             ColumnSchema columnSchema = mergeAnalysis.getDataColumnSchemas().get(columnIndex);
+            Integer fieldNumber = requireNonNull(mergeAnalysis.getColumnHandleFieldNumbers().get(dataColumnHandle), "Field number for ColumnHandle is null");
```

Comment:
`int fieldNumber`

Reply:
Also, remove `requireNonNull`. The null check will be handled by the unboxing of the Integer.